packages/google_sign_in/tizen/src/secure_storage.cc

```diff
@@ -0,0 +1,177 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "secure_storage.h"
+
+#include <ckmc/ckmc-manager.h>
+#include <ckmc/ckmc-type.h>
+
+#include <cassert>
+
+namespace {
+
+// This value must match `_kIvSizeBytes` in `secure_storage.dart`.
+const size_t kIvSizeBytes = 12;
+
+const char *kAesKey = "AesKey";
+
+const ckmc_policy_s kDefaultKeyPolicy = {
+    .password = nullptr,
+    .extractable = false,
+};
+
+const ckmc_policy_s kDefaultDataPolicy = {
+    .password = nullptr,
+    .extractable = true,
+};
+
+enum class NameType {
+  kKey,
+  kData,
+};
+
+std::vector<std::string> GetNames(NameType name_type) {
+  ckmc_alias_list_s *ckmc_alias_list;
+  switch (name_type) {
+    case NameType::kKey:
+      ckmc_get_key_alias_list(&ckmc_alias_list);
+      break;
+    case NameType::kData:
+      ckmc_get_data_alias_list(&ckmc_alias_list);
+      break;
+    default:
+      assert(false);
+  }
+
+  std::vector<std::string> names;
+  ckmc_alias_list_s *current = ckmc_alias_list;
+  while (current != nullptr) {
+    std::string name = current->alias;
+    names.push_back(name.substr(name.find(' ') + 1));
```

Comment:
Application id is prefixed when aliases are stored. for example, if I save `AesKey`, it would be saved as `com.example.google_sign_in_tizen_example AesKey`. I'll leave a comment here.